src/Containers/FeaturedProfilesCard.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import styled from 'styled-components';
+import { AddUser } from '@styled-icons/entypo/AddUser';
+import { flex } from '../Utils/Mixins';
+
+const PROFILE_PICTURE_LIMIT = 3;
+const PROFILE_INITIALS_START_INDEX = 2;
+const PROFILE_INITIALS_END_INDEX = 5;
+const REMAINING_PROFILES_INDEX = 5;
+const MAX_PROFILES = 7;
+const START_FROM_PROFILE_INDEX = 0;
+
+export interface IProfile {
+    image: string;
+    initials: string;
+    id: number;
+}
+
+export interface IFeaturedProfilesCardProps {
+    profileData: IProfile[];
+    alt: string;
+    width?: number;
+    height?: number;
+}
+
+export const FeaturedProfilesCard: React.FC<IFeaturedProfilesCardProps> = ({
+    alt,
+    profileData,
+    width = 100,
+    height = 100,
+}): React.ReactElement => {
+    const getProfileCircles = (): any => {
+        const userProfilePicture = (profile: any): React.ReactElement => {
+            return (
+                <CircleImage background="none" key={profile.image}>
+                    <img
+                        src={profile.image}
+                        alt={alt}
+                        height={height}
+                        width={width}
+                    />
+                </CircleImage>
+            );
+        };
+
+        const userInitials = (profile: any): React.ReactElement => {
+            return (
+                <CircleImage background="orange">
+                    <CircleContent>{profile.initials}</CircleContent>
+                </CircleImage>
+            );
+        };
+
+        const remainingUserProfiles = (): React.ReactElement => {
+            const remaining_profiles_value = profileData.length - 5;
+
+            return (
+                <CircleImage background="grey">
+                    <CircleContent>{remaining_profiles_value}</CircleContent>
+                </CircleImage>
+            );
+        };
+
+        const addUserIcon = (): React.ReactElement => {
+            return (
+                <CircleImage background="grey">
+                    <Icon as={AddUser} />
+                </CircleImage>
+            );
+        };
+
+        return profileData
+            .map(
+                (profile, index): React.ReactElement => {
+                    if (index < PROFILE_PICTURE_LIMIT) {
+                        return userProfilePicture(profile);
+                    }
+                    if (
+                        index > PROFILE_INITIALS_START_INDEX &&
+                        index < PROFILE_INITIALS_END_INDEX
+                    ) {
+                        return userInitials(profile);
+                    }
+                    if (index === REMAINING_PROFILES_INDEX) {
+                        return remainingUserProfiles();
+                    }
+                    return addUserIcon();
+                },
+            )
+            .splice(START_FROM_PROFILE_INDEX, MAX_PROFILES);
```

Comment:
This logic here needs some fixes. 

Creating a excess addUserIcons to splice them off is not good. Would like to see additional logic for handling the length of for the remainingUserProfiles function, and then splicing just what you need before mapping.

The case here which isn't handled is what if I have 5 or less profiles?

Reply:
bump.